model/project_ref.go

```diff
@@ -2245,54 +2250,78 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
 }
 
 // ValidateProjectCreation returns a boolean if you should surface the error or not.
-func ValidateProjectCreation(projectId string, config *evergreen.Settings, projectRef *ProjectRef) (bool, error) {
+func ValidateProjectCreation(projectId string, config *evergreen.Settings, projectRef *ProjectRef) (int, error) {
 	if config.ProjectCreation.TotalProjectLimit == 0 || config.ProjectCreation.RepoProjectLimit == 0 {
-		return false, nil
+		return 0, nil
 	}
-	catcher := grip.NewBasicCatcher()
-	allEnabledProjects, err := GetEnabledProjects()
+	originalRef, err := findOneProjectRefQ(byId(projectId))
 	if err != nil {
-		return true, errors.Wrap(err, "getting number of projects")
+		return http.StatusInternalServerError, errors.Wrapf(err, "getting project '%s'", projectId)
 	}
-	allEnabledProjectsId := []string{}
-	for _, p := range allEnabledProjects {
-		allEnabledProjectsId = append(allEnabledProjectsId, p.Id)
+	if originalRef != nil {
+		if projectRef == nil {
+			// Should be unreachable
+			return http.StatusInternalServerError, errors.New("cannot create project while defaulting to repo")
+		}
+		originalRef, err = GetProjectRefMergedWithRepo(*originalRef)
+		if err != nil {
+			return http.StatusInternalServerError, errors.Wrapf(err, "getting merged project for '%s'", originalRef.Id)
+		}
 	}
-	// No need to validate if project was enabled to begin with.
-	if utility.StringSliceContains(allEnabledProjectsId, projectId) {
-		return false, nil
+	var newRef *ProjectRef
+	if projectRef == nil {
+		// Default to repo case
+		repoRef, err := FindOneRepoRef(originalRef.Id)
+		if err != nil {
+			return http.StatusInternalServerError, errors.Wrapf(err, "getting repo ref for '%s'", originalRef.Id)
+		}
+		if repoRef == nil {
+			return http.StatusInternalServerError, errors.Errorf("no repo ref for '%s'", originalRef.Id)
+		}
+		newRef.Enabled = repoRef.Enabled
+		newRef.Owner = repoRef.Owner
+		newRef.Repo = repoRef.Repo
+	} else {
+		mergedRef, err := GetProjectRefMergedWithRepo(*projectRef)
+		if err != nil {
+			return http.StatusInternalServerError, errors.Wrapf(err, "getting merged project for '%s'", projectId)
+		}
+		newRef = mergedRef
 	}
 
-	// Only error if we are enabling a previously disabled project.
-	shouldError := false
-	pRef, err := findOneProjectRefQ(byId(projectId))
-	if err != nil {
-		return true, errors.Wrapf(err, "getting project '%s", projectId)
-	}
-	if pRef != nil {
-		// Defaulting to repo
-		if projectRef == nil {
-			projectRef, err = GetProjectRefMergedWithRepo(*pRef)
+	catcher := grip.NewBasicCatcher()
+	if originalRef == nil || newRef.IsEnabled() {
```

Comment:
These if statements need some rethinking or commenting. What is going on here? Why do we check if originalRef == nil in both if statements? Does that imply "defaultToRepo"? We should set that at the top instead to make it way clearer what's going on.

Reply:
the original == nil checks if the project previously exists or not and that means we're creating/copying

I don't think putting it on the top works because there's an extra condition for checking the owner/repo and that would duplicate more code. I put in comments to make this clearer